atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraRepairEteTest.java

```diff
@@ -0,0 +1,164 @@
+    private static final String NAMESPACE = "ns";
+    private static final String TABLE_1 = "table1";
+    private static final TableReference TABLE_REF =
+            TableReference.create(com.palantir.atlasdb.keyvalue.api.Namespace.create(NAMESPACE), TABLE_1);
+
+    private CassandraRepairHelper cassandraRepairHelper;
+    private CassandraKeyValueService kvs;
+    private CassandraKeyValueServiceConfig config;
+
+    @Before
+    public void setUp() {
+        MetricsManager metricsManager =
+                new MetricsManager(new MetricRegistry(), new DefaultTaggedMetricRegistry(), _unused -> true);
+
+        config = ThreeNodeCassandraCluster.getKvsConfig(2);
+        kvs = CassandraKeyValueServiceImpl.createForTesting(config);
+
+        kvs.createTable(TABLE_REF, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        kvs.putUnlessExists(TABLE_REF, ImmutableMap.of(NONEMPTY_CELL, CONTENTS));
+
+        cassandraRepairHelper = new CassandraRepairHelper(metricsManager, _unused -> config, _unused -> kvs);
+    }
+
+    @After
+    public void tearDown() {
+        kvs.dropTable(TABLE_REF);
+    }
+
+    @Test
+    public void shouldGetRangesForBothReplicas() {
+        Map<InetSocketAddress, Set<LightweightOppTokenRange>> ranges =
+                cassandraRepairHelper.getRangesToRepair(Namespace.of(NAMESPACE), TABLE_1);
+        assertThat(ranges).hasSize(2);
+    }
+
+    @Test
+    public void tokenRangesToRepairShouldBeSubsetsOfTokenMap() {
+        Map<InetSocketAddress, Set<LightweightOppTokenRange>> fullTokenMap = getFullTokenMap();
+        Map<InetSocketAddress, Set<LightweightOppTokenRange>> rangesToRepair =
+                cassandraRepairHelper.getRangesToRepair(Namespace.of(NAMESPACE), TABLE_1);
+
+        KeyedStream.stream(rangesToRepair)
+                .forEach((address, cqlRangesForHost) ->
+                        assertRangesToRepairAreSubsetsOfRangesFromTokenMap(fullTokenMap, address, cqlRangesForHost));
+    }
+
+    // The ranges in CQL should be a subset of the Thrift ranges, except that the CQL ranges are also snipped,
+    // such that if the thrift range is [5..9] but we don't have data after 7, then the CQL range will be [5..7]
+    private void assertRangesToRepairAreSubsetsOfRangesFromTokenMap(
+            Map<InetSocketAddress, Set<LightweightOppTokenRange>> fullTokenMap,
+            InetSocketAddress address,
+            Set<LightweightOppTokenRange> cqlRangesForHost) {
+        String hostName = address.getHostName();
+        InetSocketAddress thriftAddr = new InetSocketAddress(hostName, MultiCassandraUtils.CASSANDRA_THRIFT_PORT);
+        Set<LightweightOppTokenRange> thriftRangesForHost = fullTokenMap.get(thriftAddr);
+        assertThat(thriftRangesForHost).isNotNull();
+        cqlRangesForHost.forEach(range -> assertThat(thriftRangesForHost.stream()
+                        .anyMatch(thriftRange -> thriftRange.left().equals(range.left())))
```

Comment:
Also check that the right side is less or equal to (taking into account wrap around)?